workflow/controller/taskset_test.go

```diff
@@ -333,3 +334,175 @@ func TestNonHTTPTemplateScenario(t *testing.T) {
 		assert.NoError(t, err)
 	})
 }
+
+func TestReconcileTaskSetWithMemoization(t *testing.T) {
+	wf := wfv1.MustUnmarshalWorkflow(`apiVersion: argoproj.io/v1alpha1
+kind: Workflow
```

Comment:
This example makes it too hard to understand what is the example is relevant to the test, and what is just copied and pasted. Can you please simplify.

Reply:
ok